pkgs/by-name/nn/nng/package.nix

```
@@ -27,7 +27,10 @@ stdenv.mkDerivation rec {
   buildInputs = lib.optionals mbedtlsSupport [ mbedtls ];
 
   cmakeFlags =
-    [ "-G Ninja" ]
+    [
+      "-G Ninja"
+      "-DBUILD_SHARED_LIBS=ON"
```

```suggestion
      (lib.cmakeBool "BUILD_SHARED_LIBS" !stdenv.hostPlatform.isStatic)
```
Also, sorry to do this but I just realized this is *probably* more correct? I see both used frequently: https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+BUILD_SHARED_LIBS&type=code

---

tbh, I am not very experienced with the build options of NNG. I added the package to fulfill a dependency of the nanonext, mirai, and crew libraries in R - HPC tools. I'll try building those libraries with this change to confirm nothing breaks downstream.

---

That seems like a good change, I do not think it'll break anything
It does still parenthesis around the negation though - like this
`(lib.cmakeBool "BUILD_SHARED_LIBS" (!stdenv.hostPlatform.isStatic))`

nng builds both shared and static with the expected results

both the R packages still build on my machine (only x86_64) with this change

SatDump also still builds & runs